### PR TITLE
Support labels/annotations in sidecar task

### PR DESCRIPTION
**Pull request by milton0825**
# TL;DR
Support specifying pod labels/annotations in sidecar task

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
NA

## Follow-up issue
NA

### CODE REVIEW

**Comment by milton0825** [comment]
@anandswaminathan @akhurana001

**Review by kumare3 — COMMENTED** [review]
can you please add a unit test?

File: go/tasks/plugins/k8s/sidecar/sidecar.go [hunk 1]
```diff
@@ -71,11 +70,15 @@ func validateAndFinalizePod(
 type sidecarJob struct {
 	PodSpec              *k8sv1.PodSpec
 	PrimaryContainerName string
+	Annotations          map[string]string
```
Comment (kumare3) [inline go/tasks/plugins/k8s/sidecar/sidecar.go:73]: why do we need this, why can we not use annotations in the PodSpec directly?

Also adding this to struct will not work, it has to be marshalled from flytekit - how does that happen?

Comment (milton0825) [reply go/tasks/plugins/k8s/sidecar/sidecar.go:73]: Annotations/Labels is not part of the `PodSpec`, it is in the `Pod` .

Comment (milton0825) [reply go/tasks/plugins/k8s/sidecar/sidecar.go:73]: So I am planning to open a Pr in flytekit as well to allow annotations/labels for sidecar task.

**Comment by milton0825** [comment]
I have updated the existing unit tests to coverr annotations/labels

File: go/tasks/plugins/k8s/sidecar/sidecar.go (hunk 1 above)
Comment (milton0825) [reply go/tasks/plugins/k8s/sidecar/sidecar.go:73]: https://github.com/flyteorg/flyteidl/pull/171 is for the IDL change

**Review by katrogan — APPROVED** [review]

**Comment by katrogan** [comment]
mind updating the flyteidl version in go.mod with [v0.18.47]( https://github.com/flyteorg/flyteidl/releases/tag/v0.18.47)

**Comment by milton0825** [comment]
@katrogan done!

**Review by katrogan — APPROVED** [review]

**Pull request merged**